src/AbstractSQLSchemaOptimiser.ts

```diff
@@ -32,6 +32,17 @@ const countFroms = (n: AbstractSqlType[]) => {
 	return count;
 };
 
+export const generateRuleHashAcronym = (
+	tableName: string,
+	ruleBody: AbstractSqlType,
+) => {
+	const sha = sbvrTypes.SHA.validateSync(
+		`${tableName}$${JSON.stringify(ruleBody)}`,
+	).replace(/^\$sha256\$/, '');
+	// Trim the trigger to a max of 63 characters, reserving at least 32 characters for the hash
+	return `${tableName.slice(0, 30)}$${sha}`.slice(0, 63);
```

Comment:
I think it'd make sense for this to take inspiration from the roughly equivalent code in https://github.com/balena-io-modules/odata-to-abstract-sql/blob/v6.2.4/src/odata-to-abstract-sql.ts#L393-L396 - the things of note to me are:
1. using base36 allows it to still be a string but also fit as much of the hash in as possible in the same number of characters (vs hex/base 16)
2. adjusting the truncation dynamically based upon the actual strings
3. using a named constant for `MAX_ALIAS_LENGTH` rather than a magic number for which the reasoning is not as clear